src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/CollectionApiDataType.java

```diff
@@ -3,7 +3,7 @@
 import com.datastax.oss.driver.api.core.type.DataType;
 
 /** Interface defining the api data type for collection types */
-public abstract class CollectionApiDataType implements ApiDataType {
+public abstract class CollectionApiDataType<T extends DataType> implements ApiDataType {
```

Comment:
doc comments to explain the type 

also , why do we need this to be generic now ? that would be something to put in the doc comments :) 

why are we using the CQL data type ? In general these classes should be our internal data model